contracts/protocol/tokenization/base/ScaledBalanceTokenBase.sol

```diff
@@ -125,4 +125,36 @@ abstract contract ScaledBalanceTokenBase is MintableIncentivizedERC20, IScaledBa
       emit Burn(user, target, amountToBurn, balanceIncrease, index);
     }
   }
+
+  /**
+   * @notice Implements the basic logic to transfer scaled balance tokens between two users
+   * @dev It emits a mint event with the interest accrued per user 
+   * @param sender The source address
+   * @param recipient The destination address
+   * @param amount The amount getting transferred
+   * @param index The next liquidity index of the reserve
+   **/
+  function _transfer(
+    address sender,
+    address recipient,
+    uint256 amount,
+    uint256 index
+  ) internal {
+    uint256 senderScaledBalance = super.balanceOf(sender);
+    uint256 senderBalanceIncrease = senderScaledBalance.rayMul(index) -
+      senderScaledBalance.rayMul(_userState[sender].additionalData);
+
+    uint256 recipientScaledBalance = super.balanceOf(recipient);
+    uint256 recipientBalanceIncrease = recipientScaledBalance.rayMul(index) -
+      recipientScaledBalance.rayMul(_userState[recipient].additionalData);
+
+    _userState[sender].additionalData = index.toUint128();
+    _userState[recipient].additionalData = index.toUint128();
+
+    super._transfer(sender, recipient, amount.rayDiv(index).toUint128());
+
+    emit Transfer(sender, recipient, amount);
+    emit Mint(sender, sender, senderBalanceIncrease, senderBalanceIncrease, index);
```

Comment:
Maybe `Transfer` events should be emitted for the `balanceIncrease` too? The system then functions like a regular, non-rebasing ERC20 token, with all transfer events adding up to the current balance.

```solidity
emit Mint(sender, sender, senderBalanceIncrease, senderBalanceIncrease, index);
emit Transfer(address(0), sender, senderBalanceIncrease);

if (recipientBalanceIncrease > 0) {
  emit Mint(recipient, recipient, recipientBalanceIncrease, recipientBalanceIncrease, index);
  emit Transfer(address(0), recipient, recipientBalanceIncrease);
}

emit Transfer(sender, recipient, amount);
```

Reply:
Good points, both addressed. Thanks